### PR TITLE
Bump the version of kubernetes-cni to 0.6.0

### PR DESCRIPTION
**Pull request by mauilion**
This will resolve the kubernetes-anywhere e2e test.
In commit: https://github.com/kubernetes/kubernetes/pull/71540
I bumped the required version of kubernetes-cni to 0.6.0 but didn't
start packaging it. This resolve that.

Signed-off-by: Duffie Cooley <dcooley@heptio.com>

**What type of PR is this?**
/kind failing-test

**What this PR does / why we need it**:
This bumps the version of kubernetes-cni that we package to 0.6.0

**Special notes for your reviewer**:
This is related to: https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-e2e-kubeadm-gce-master/1018/artifacts/e2e-1018-1ad8f-master/serial-1.log

**Does this PR introduce a user-facing change?**:
```release-note
Bumps version of kubernetes-cni to 0.6.0
```

### CODE REVIEW

**Comment by mauilion** [comment]
For context: https://kubernetes.slack.com/archives/C13J86Z63/p1543642853032600

**Comment by AishSundar** [comment]
Can we please hold this until Monday 12/3 so that we can get 1.13 out without any CI signal disruption. Let me know if there's any urgency for this to go in before that.

/hold

**Comment by mauilion** [comment]
It won't. This just addresses master iiuc. It will fix the failing k/anywhere test tho.

**Comment by timothysc** [comment]
/hold cancel 

@AishSundar - this is for master/1.14 only

**Comment by timothysc** [comment]
@mauilion - how can we verify this fixes.  I'm missing how 0.5.1 was installed on k8s-anywhere.

File: build/BUILD
Comment (timothysc) [inline build/BUILD]: This means we have multiple locations for CNI version.  

Can we bake it into a single variable?

Comment (mauilion) [reply build/BUILD]: Per our conversation. This is fine to merge as it stands @timothysc but we need to capture the requirement that the dependent version and the built version are locked to the same var.

Comment (ixdy) [reply build/BUILD]: yes, this is what we do with the CRI version:
https://github.com/kubernetes/kubernetes/blob/8c0542dcf126222481390cd0606e3b7f43830a64/build/workspace.bzl#L15

we should do something similar here.

Comment (ixdy) [reply build/BUILD]: though then also we should probably use that variable in the `WORKSPACE`:

https://github.com/kubernetes/kubernetes/blob/8c0542dcf126222481390cd0606e3b7f43830a64/build/root/WORKSPACE#L57-L62

**Comment by timothysc** [comment]
> AFAICT, it's been 0.6.0 for a long time, with the package version not matching ... 

It's pretty clear from the logs that it was not doing that, but was running 0.5.1

**Review by timothysc — APPROVED** [review]
/lgtm 
/approve 

I'm going to unblock this to try and get master green and open some tracking issues to fix some of these issues.

**Comment by mauilion** [comment]
> > AFAICT, it's been 0.6.0 for a long time, with the package version not matching ...
> 
> It's pretty clear from the logs that it was not doing that, but was running 0.5.1

What's happening here is even more messed up tho. Basically bazel would pull version 0.6.0 (because of the WORKSPACE config) and then "label" it 0.5.1 (becaus

**Comment by mauilion** [comment]
> > AFAICT, it's been 0.6.0 for a long time, with the package version not matching ...
> 
> It's pretty clear from the logs that it was not doing that, but was running 0.5.1

What's happening here is even more messed up tho. Basically bazel would pull version 0.6.0 (because of the WORKSPACE config) and then "label" it 0.5.1 (because of the k/k/build/BUILD configuration)

So when you install the resulting rpm it is holding version 0.6.0 but labeled 0.5.1...

**Comment by cblecker** [comment]
@mauilion Can you take a look at @ixdy's recommended changes above?

**Comment by cblecker** [comment]
SGTM 👍 

/lgtm
/approve

**Comment by BenTheElder** [comment]
follow up here: https://github.com/kubernetes/kubernetes/pull/71686

**Pull request merged**
